realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -351,6 +414,60 @@ private void clearScheduledAccessTokenRefresh() {
         }
     }
 
+    // Wrapper class for handling the async operations of the underlying SyncSession calling `async_wait_for_download_completion`
+    private static class WaitForServerChangesWrapper {
+
+        private final CountDownLatch waitForChanges = new CountDownLatch(1);
+        private volatile boolean resultReceived = false;
+        private Long errorCode = null;
+        private String errorMessage;
+
+        public void cancelWait() {
```

Comment:
This is not called by anyone I think? Actually this can just be removed. `getInstanceAsync()` will be using `RealmAsyncTask`, when calling `RealmAsyncTask.cancel()`, `CountDownLatch.await()` will just throw a `InterruptedException`. see line 436 in this file. Also, line 437 should log the exception instead of throw it.